collector/set.go

```diff
@@ -69,6 +69,29 @@ func (s *Set) Boot(ctx context.Context) error {
 	return nil
 }
 
+func (s *Set) Stop(ctx context.Context) {
+	s.logger.Log("level", "debug", "message", "stopping collector")
+
+	if !s.isBooted() {
+		return
+	}
+
+	{
+		s.logger.LogCtx(ctx, "level", "debug", "message", "unregistering collector")
+
+		registered := prometheus.Unregister(s)
+		if !registered {
+			s.logger.LogCtx(ctx, "level", "debug", "message", "collector was not registered")
+		} else {
+			s.logger.LogCtx(ctx, "level", "debug", "message", "unregistered collector")
+		}
+	}
+
+	s.logger.Log("level", "debug", "message", "stopped collector")
+
```

Comment:
I'm not certain about the value of those logs but let's keep them if you think they are legit.

Reply:
It's pretty verbose, but this should only be executed rarely, and I could see it helping with future debugging. No strong feelings though.